client/my-sites/stats/stats-purchase/stats-purchase-wizard.jsx

```diff
@@ -0,0 +1,224 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import { Button, CheckboxControl, Card, Panel, PanelRow, PanelBody } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import React, { useState } from 'react';
+import PersonalPurchase from './stats-purchase-personal';
+import StatsPurchaseSVG from './stats-purchase-svg';
+import './styles.scss';
+
+const COMPONENT_NAME = 'stats-purchase-wizard';
+const SCREEN_TYPE_SELECTION = 0;
+const SCREEN_PERSONAL_CHECKLIST = 1;
+const SCREEN_PURCHASE = 2;
+const TYPE_PERSONAL = 'Personal';
+const TYPE_COMMERCIAL = 'Commercial';
+const FLAT_COMMERTIAL_PRICE = 10;
+
+// TODO: import
+const CommercialPurchase = () => {
+	return (
+		<div>
+			<p>Notice...</p>
+
+			<p>${ `${ FLAT_COMMERTIAL_PRICE }` }/month</p>
+
+			<div className="benefits">
+				<ul className="included">
+					<li>Instant access to upcoming features</li>
+					<li>Priority support</li>
+					<li>Ad-free experience</li>
+				</ul>
+			</div>
+
+			<p>
+				By clicking the button below, you agree to our <a href="#">Terms of Service</a> and to{ ' ' }
+				<a href="#">share details</a> with WordPress.com.
+			</p>
+
+			<Button isPrimary>Get Jetpack Stats for ${ FLAT_COMMERTIAL_PRICE } per month</Button>
+			<Button isSecondary>Continue for free</Button>
+		</div>
+	);
+};
+
+const ProductCard = ( { siteSlug } ) => {
+	const [ subscriptionValue, setSubscriptionValue ] = useState( 6 );
+	const [ wizardStep, setWizardStep ] = useState( SCREEN_TYPE_SELECTION );
+	const [ siteType, setSiteType ] = useState( null );
+	const [ isAdsChecked, setAdsChecked ] = useState( false );
+	const [ isSellingChecked, setSellingChecked ] = useState( false );
+	const [ isBusinessChecked, setBusinessChecked ] = useState( false );
+	const translate = useTranslate();
+
+	const personalLabel = translate( 'Personal site' );
+	const commertialLabel = translate( 'Commercial site' );
+	const selectedTypeLabel = siteType === TYPE_PERSONAL ? personalLabel : commertialLabel;
+
+	const handleTypeClick = ( type ) => {
+		setSiteType( type );
+		setWizardStep( type === TYPE_PERSONAL ? SCREEN_PERSONAL_CHECKLIST : SCREEN_PURCHASE );
+	};
+
+	const toggleFirstStep = ( toggleState ) => {
+		// Never close on click but allow to open when a type was selected (to change).
+		if ( ! siteType || ! toggleState ) {
+			return;
+		}
+
+		setWizardStep( SCREEN_TYPE_SELECTION );
+	};
+
+	// change the plan to commertial on the personal plan confirmation
+	const handlePlanSwap = () => {
+		setSiteType( TYPE_COMMERCIAL );
+		setWizardStep( SCREEN_PURCHASE );
+	};
+
+	return (
+		<div className={ COMPONENT_NAME }>
+			<Card className="jetpack-upsell-card">
+				<div className={ `${ COMPONENT_NAME }__card` }>
+					<div className="left">
+						<Panel className={ `${ COMPONENT_NAME }__card-panel` } header="Jetpack Stats">
+							<PanelBody
```

Comment:
For a follow-up PR: It might be good to refactor this into a separate component (StatsSiteTypeSelector, maybe?)